estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,70 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare
+
+
+class PropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Real Estate Property Offer'
+    _order = 'price desc'
+
+    # Custom fields
+    price = fields.Float()
+    status = fields.Selection([('accepted', 'Accepted'), ('refused', 'Refused')], copy=False)
+    partner_id = fields.Many2one('res.partner', required=True)
+    property_id = fields.Many2one('estate.property', required=True)
+
+    # Constraints
+    _sql_constraints = [
+        ('check_price', 'CHECK(price >= 0)', 'The offer price must be positive.'),
+    ]
+
+    # Computed fields
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute='_compute_date_deadline', inverse='_inverse_date_deadline')
+
+    @api.depends('create_date', 'validity')
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                date = record.create_date
+            else:
+                date = fields.Date.today()
```

Comment:
```suggestion
           date = record.create_date or fields.Date.today()
```
Another solution than the slides